crates/bevy_ecs/src/world/mod.rs

```diff
@@ -1324,7 +1473,7 @@ impl World {
     unsafe fn initialize_resource_internal(
```

Comment:
This should have a safety invariant that `component_id` is send or else we're going to be creating send storage for a !Send type and skipping the checks